src/transformers/models/oneformer/processing_oneformer.py

```diff
@@ -108,76 +165,72 @@ def __call__(self, images=None, task_inputs=None, segmentation_maps=None, **kwar
-            raise ValueError("You have to specify the image. Found None.")
-
-        if not all(task in ["semantic", "instance", "panoptic"] for task in task_inputs):
-            raise ValueError("task_inputs must be semantic, instance, or panoptic.")
-
-        encoded_inputs = self.image_processor(images, task_inputs, segmentation_maps, **kwargs)
-
+        output_kwargs = self._merge_kwargs(
+            OneFormerProcessorKwargs,
+            tokenizer_init_kwargs=self.tokenizer.init_kwargs,
+            **kwargs,
+            **self._add_args_for_backward_compatibility(args),
+        )
+        segmentation_maps = output_kwargs["images_kwargs"].pop("segmentation_maps", None)
+        task_inputs = output_kwargs["images_kwargs"].pop("task_inputs", None)
         if isinstance(task_inputs, str):
             task_inputs = [task_inputs]
+        self._validate_input_types(images, task_inputs)
 
-        if isinstance(task_inputs, List) and all(isinstance(task_input, str) for task_input in task_inputs):
-            task_token_inputs = []
-            for task in task_inputs:
-                task_input = f"the task is {task}"
-                task_token_inputs.append(task_input)
-            encoded_inputs["task_inputs"] = self._preprocess_text(task_token_inputs, max_length=self.task_seq_length)
-        else:
-            raise TypeError("Task Inputs should be a string or a list of strings.")
-
-        if hasattr(encoded_inputs, "text_inputs"):
-            texts_list = encoded_inputs.text_inputs
+        encoded_inputs = self.image_processor(images, task_inputs, segmentation_maps, **output_kwargs["images_kwargs"])
 
-            text_inputs = []
-            for texts in texts_list:
-                text_input_list = self._preprocess_text(texts, max_length=self.max_seq_length)
-                text_inputs.append(text_input_list.unsqueeze(0))
+        task_token_inputs = []
+        for task in task_inputs:
+            task_input = f"the task is {task}"
+            task_token_inputs.append(task_input)
+        encoded_inputs["task_inputs"] = self._preprocess_text(
+            task_token_inputs,
+            max_length=output_kwargs["text_kwargs"]["task_seq_length"],
+        )
 
+        if hasattr(encoded_inputs, "text_inputs"):
+            text_inputs = [
+                self._preprocess_text(
+                    texts,
+                    max_length=output_kwargs["text_kwargs"]["max_seq_length"],
+                ).unsqueeze(0)
+                for texts in encoded_inputs.text_inputs
+            ]
             encoded_inputs["text_inputs"] = torch.cat(text_inputs, dim=0)
 
         return encoded_inputs
 
-    def encode_inputs(self, images=None, task_inputs=None, segmentation_maps=None, **kwargs):
+    def encode_inputs(
+        self,
+        images=None,
+        task_inputs=None,
+        segmentation_maps=None,
+        max_seq_length: int = 77,
+        task_seq_length: int = 77,
+        **kwargs,
+    ):
```

Comment:
With this refactoring, it looks to me like encode_inputs could be made the same as __call__ while preserving backward compatibility (if the args are handled correctly), so maybe we could just use __call__ here?

Reply:
Good catch! Done.